File: result/translate/chatgpt/java2cpp/c++/OVERLAPPING_SUM_TWO_ARRAY.cpp
#include <unordered_map>

int f_gold(int A[], int B[], int n) {
    std::unordered_map<int, int> hash;
    for (int i = 0; i < n; i++) {
        if (hash.count(A[i]))
            hash[A[i]] = 1 + hash[A[i]];
        else
            hash[A[i]] = 1;
        
        if (hash.count(B[i]))
            hash[B[i]] = 1 + hash[B[i]];
        else
            hash[B[i]] = 1;
    }
    
    int sum = 0;
    for (auto entry : hash) {
        if (std::stoi(std::to_string(entry.second)) == 1)
            sum += std::stoi(std::to_string(entry.first));
    }
    
    return sum;
}

File: result/translate/transcoder/python2java/java/OVERLAPPING_SUM_TWO_ARRAY.java
public static int f_gold ( int [ ] A , int [ ] B , int n ) {
  TreeMap < Integer , Integer > Hash = new TreeMap < Integer , Integer > ( ) ;
  for ( int i = 0 ;
  i <= n ;
  i ++ ) {
    Hash . put ( A [ i ] , 0 ) ;
    Hash . put ( B [ i ] , 1 ) ;
  }
  int Sum = 0 ;
  for ( int x : Hash . keySet ( ) ) {
    if ( Hash . get ( x ) == 1 ) Sum += x ;
  }
  return Sum ;
}